opentaskspal/src/main/java/org/dmfs/opentaskspal/tasks/NoTimeData.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * {@link RowData} for clearing time data (start, due, duration) for a task.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class NoTimeData implements RowData<TaskContract.Tasks>
+{
+    @NonNull
+    @Override
+    public ContentProviderOperation.Builder updatedBuilder(@NonNull TransactionContext transactionContext, @NonNull ContentProviderOperation.Builder builder)
+    {
+        return builder
+                .withValue(TaskContract.Tasks.DTSTART, null)
+                .withValue(TaskContract.Tasks.DUE, null)
+                .withValue(TaskContract.Tasks.DURATION, null);
+    }
```

Comment:
A task without times is currently not allowed to recur, so RDATE, RRULE and EXDATE should also be set to null.

Reply:
Also wipe out all-day and time zone.

Reply:
Updated.